### ISSUE TITLE
Encapsulation Key Random Byte

### ISSUE THREAD

**Issue by jmwample** [jmwample/kemeleon#17]
TLDR:  the one random byte needed for the encapsulation key causes problems

the Kemeleon `EncapsulationKey` needs one extra random byte to make the high order byte fully random. When the DECAPSULATION key is serialized then de-serialized (i.e. written to file then read back) you need to recover (or re-derive) the encapsulation key from the decapsulation key itself. HOWEVER, that byte is not written out with the key in any way. So the re-derived ML-KEM encapsulation key may not have the same upper bits (when Kemeleon encoded) as before.

The extra byte has to be stored with the decapsulation key too and written out to / read from wherever we are serializing to.

**Comment by jmwample** [comment]
Note: As of now there isn't even a good way to wrap `ml_kem::kem::DecapsulationKey<P>` as the required trait `KemParams` is private. So you have to go through the `KemCore::DecapsulationKey` trait as a proxy, but this doesn't allow you to treat it as a struct - only as a trait object which implements the things required of the associated type.  So you don't get access to `DecapsulationKey::encapsulation_key()`.

**Comment by jmwample** [comment]
For now I have made the decision to use hkdf(dk.as_bytes(), ":static_mac_val", Null) to make sure the byte is deterministically generated. This may not be a permanent solution, but it should work for now. #21
